Doesn't build on riscv32i-unknown-none-elf without default features (regression 0.7.10)

When adding `heapless` as a dependency:
```toml
[dependencies]
heapless = { version = "=0.7.10", default-features = false, features = [] }
```
it does not build:
```
   Compiling heapless v0.7.10
error[E0432]: unresolved import `atomic_polyfill`
  --> /home/kriskras99/.cargo/registry/src/github.com-1ecc6299db9ec823/heapless-0.7.10/src/spsc.rs:90:5
   |
90 | use atomic_polyfill::{AtomicUsize, Ordering};
   |     ^^^^^^^^^^^^^^^ use of undeclared crate or module `atomic_polyfill`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `heapless` due to previous error
```
This regression was introduced with version `0.7.10` (aka it works with `version = "=0.7.9"`)
Also see [the Rust Crypto issue](https://github.com/RustCrypto/AEADs/issues/398).

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -53,6 +53,7 @@ jobs:
           - nightly
         features:
           - ""
+          - "cas"
           - "serde"
     steps:
       - name: Checkout
@@ -92,7 +93,7 @@ jobs:
         with:
           use-cross: false
           command: check
-          args: --target=${{ matrix.target }} --features=${{ matrix.features }}
+          args: --target=${{ matrix.target }} --no-default-features --features=${{ matrix.features }}
 
   # Run cpass tests
   testcpass:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,6 +19,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 
 * Optimize the codegen of `Vec::clone`
+* `riscv32i` targets unconditionally (`no-default-features`) depends on `atomic-polyfill`
 
 ### Fixed
 

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -19,7 +19,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ### Changed
 
 * Optimize the codegen of `Vec::clone`
-* `riscv32i` targets unconditionally (`no-default-features`) depends on `atomic-polyfill`
+* `riscv32i` and `riscv32imc` targets unconditionally (e.g. `build --no-default-features`) depends on `atomic-polyfill`
 
 ### Fixed
 

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,7 +31,7 @@ defmt-impl = ["defmt"]
 atomic-polyfill = { version = "0.1.2", optional = true }
 
 [target.riscv32i-unknown-none-elf.dependencies]
-atomic-polyfill = { version = "0.1.4", optional = true }
+atomic-polyfill = { version = "0.1.4" }
 
 [target.riscv32imc-unknown-none-elf.dependencies]
 atomic-polyfill = { version = "0.1.4", optional = true }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -34,7 +34,7 @@ atomic-polyfill = { version = "0.1.2", optional = true }
 atomic-polyfill = { version = "0.1.4" }
 
 [target.riscv32imc-unknown-none-elf.dependencies]
-atomic-polyfill = { version = "0.1.4", optional = true }
+atomic-polyfill = { version = "0.1.4" }
 
 [dependencies]
 hash32 = "0.2.1"

